pkg/custom-provider/provider.go

```diff
@@ -146,8 +205,15 @@ func (p *prometheusProvider) buildQuery(ctx context.Context, info provider.Custo
 
 	// TODO: use an actual context
 	queryResults, err := p.promClient.Query(ctx, pmodel.Now(), query)
+	re := regexp.MustCompile(`\[Status Code: (\d{3})\]`)
 	if err != nil {
 		klog.Errorf("unable to fetch metrics from prometheus: %v", err)
+		matches := re.FindStringSubmatch(err.Error())
+    statusCode := "unknown"
+    if len(matches) > 1 {
+      statusCode = matches[1] // The captured status code
+    }
```

Comment:
Is there a way to do this without regex? The regex way is a bit hacky in the sense that we can't always guarantee the status code will be in that exact format